app/frontend/src/pages/encrypt/Encrypt.page.tsx

```diff
@@ -9,13 +14,22 @@ export const EncryptPage = (): JSX.Element => {
   const [password, setPassword] = useState('');
   const [link, setLink] = useState('');
 
+  const searchParams = useSearchParams();
+  const id = searchParams.get<string>('id');
+
   const handlePasswordChange = (newPassword: string) => {
     setPassword(newPassword);
     setLink('');
   };
 
-  const handleLinkGeneration = () => {
-    setLink(Math.random().toString());
+  const handleLinkGeneration = async (newPassword: string) => {
+    if (id) {
+      const response = await post<EncryptionResponse, EncryptionParams>('/encrypt', { id, password: newPassword });
+      if (response.message === 'OK') {
+        const { id, secret } = response as { id: string; secret: string };
+        setLink(`localhost:3000/decrypt?id=${id}&secret=${secret}`);
```

Comment:
Please move 'localhost:3000' to some environment variable (since our hostname will change).